app/models/template.rb

```
@@ -126,8 +137,11 @@ def self.parse_metadata(text)
   #   :lock - lock imported templates (false by default), can be either boolean or lambda
   #   :default - default flag value (false by default)
   def self.import!(name, text, options = {})
-    template = import_without_save(name, text, options)
-    return template unless template.valid?
+    template = import_without_save(name, text, options) if valid_params?(parse_metadata(text))
+    if template.valid?
+      template.save!
+      return template
+    end
```

Why we are addin these? It would seem that the lines below got it covered already.

---

the validation was wrong from what I saw . 
It was: 
` return template unless template.valid?`
and the template wasn't saved.

---

isn't it correct? In case the template is invalid, we should not try to save it because it would fail anyway, or am I missing something?